### PR TITLE
make _siblingIndex/_parent to protected

### PR DESCRIPTION
**Pull request by minggo**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/scene-graph/node.ts [hunk 1]
```diff
@@ -336,11 +336,7 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
         }
     }
 
-    /**
-     * @engineInternal this is engineInternal for it doesn't have side effect of setting parent.
-     */
-    @serializable
-    public _parent: this | null = null;
+    protected _parent: this | null = null;
```
Comment (dumganhar) [inline cocos/scene-graph/node.ts:340]: Why remove the `@serializable` decorator ?

File: cocos/scene-graph/prefab/utils.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export function createNodeWithPrefab (node: Node) {
 
     // restore preserved props
     node._objFlags = _objFlags;
-    node._parent = _parent;
+    node.parent = _parent;
```
Comment (dumganhar) [inline cocos/scene-graph/prefab/utils.ts]: This change modify the original logic `node.parent = _parent` setter will do more logic than `node._parent = parent`.

```js
set parent (value) {
        this.setParent(value);
    }
```

File: cocos/scene-graph/node.ts (hunk 1 above)
Comment (minggo) [reply cocos/scene-graph/node.ts:340]: It is a mistake. Add it back.

File: cocos/scene-graph/prefab/utils.ts (hunk 2 above)
Comment (minggo) [reply cocos/scene-graph/prefab/utils.ts]: Yep. I should fix it. It is wired that setter of `_parent` has the same logic as `setParent`.

File: cocos/scene-graph/node.jsb.ts [hunk 3]
```diff
@@ -998,17 +998,33 @@ Object.defineProperty(nodeProto, '_siblingIndex', {
     configurable: true,
     enumerable: true,
     get() {
-        return this._sharedInt32Arr[0]; // Int32, 0: siblingIndex
+        return this.getSiblingIndex(this);
```
Comment (PPpro) [inline cocos/scene-graph/node.jsb.ts]: why getSiblingIndex has a this argument

Comment (minggo) [reply cocos/scene-graph/node.jsb.ts]: Yep, this is not needed.

Comment (minggo) [reply cocos/scene-graph/node.jsb.ts]: Fixed.

File: cocos/scene-graph/prefab/utils.ts (hunk 2 above)
Comment (minggo) [reply cocos/scene-graph/prefab/utils.ts]: I added `Node.modifyParent()` to modify _parent value only.

File: cocos/scene-graph/prefab/utils.ts [hunk 4]
```diff
@@ -187,7 +187,7 @@ export function applyMountedChildren (node: Node, mountedChildren: MountedChildr
                     }
 
                     target.children.push(childNode);
-                    childNode._parent = target;
+                    childNode.parent = target;
```
Comment (PPpro) [inline cocos/scene-graph/prefab/utils.ts]: should this be `modifyParent` ?

Comment (minggo) [reply cocos/scene-graph/prefab/utils.ts]: Good catch. Fixed.

**Review by PPpro — APPROVED** [review]

File: cocos/scene-graph/prefab/utils.ts (hunk 2 above)
Comment (dumganhar) [reply cocos/scene-graph/prefab/utils.ts]: How about naming it with `changeParent` or `replaceParent`?
`modifyParent` seems to modify some content of its parent, not just replace it or change it.

Comment (minggo) [reply cocos/scene-graph/prefab/utils.ts]: Modify is more suitable. I got the name from chatgpt.

**Review by dumganhar — APPROVED** [review]

**Pull request merged**